kratos/spatial_containers/bounding_volume_tree.h

```diff
@@ -270,11 +270,11 @@ class _6DOP : public kDOP // AABB
     KRATOS_CLASS_POINTER_DEFINITION(_6DOP);
     _6DOP() : kDOP() {this->Initialize();}
     virtual ~_6DOP() {}
-    virtual std::size_t NumberOfDirections() const {return 3;}
+    virtual std::size_t NumberOfDirections() const override {return 3;}
 
 private:
     static const double msDirection[][3];
-    virtual const double (*Direction() const)[3];
+    virtual const double (*Direction() const override)[3];
```

Comment:
@roigcarlo, you are correct, virtual const double (*Direction() const)[3]; returns a set of direction vectors associated with a kDOP under the form of a 2D array. Those vectors are static so I kept an internal static array and give a function to return a pointer to it. Technically returns a const pointer to the first array of the 2D array.

The original code does not have override and it was compiled fine with gcc. But I can try your suggestion to remove the annoying of other compilers.

@pooyan-dadvand i'll take some time to check this